Cargo.toml

```diff
@@ -18,3 +18,6 @@ repository = "https://github.com/iron-fish/ironfish"
 
 [patch.crates-io]
 bellman = { git = "https://github.com/iron-fish/bellman", rev = "1cc52ca33e6db14233f1cbc0c9c5b7c822b229ec" }
+
```

Comment:
Are we ok with setting our workspace level debug info to be included? 

https://doc.rust-lang.org/cargo/reference/profiles.html#debug

Reply:
This will increase the size of the compiled binary (it shouldn't have any impact on performance though).

On Linux distributions (like Ubuntu or Debian), what we do is that we ship by default binaries without debug symbols, to keep the installed size small, and provide optional `-dbg` or `-dbgsym` packages so that people can get the debug symbols if they want to. For example there's a `libc6` package and a `libc6-dbg` package for that reason.

Cargo supports a similar mechanism through the [split-debuginfo](https://doc.rust-lang.org/cargo/reference/profiles.html#split-debuginfo) option, although I don't know if it works with napi, or even if it works in general on non-Linux (non-ELF) system. Maybe it's worth a try.

It may also be worth seeing if Node.js ships with debug symbols or not on OSes like macOS: if they do, then I would say we are "justified" to do it as well, as it would be a common practice.

Reply:
Another thing to note is that these debug symbols can be stripped at any time, so we can tell Cargo/rustc to insert debug symbols by default, but then strip them before we make official releases. This way, when you test things locally, you get debug syms, when you install the official release, you don't (saving storage space).

Reply:
These both seem like great options, honestly I think the first sounds nicer from a developer perspective, but the later is simpler to implement.


Although I am not sure for our project that it matters if we include them by default.
With debug off, our binary is 97.2 MB, with debug on it is 98.3 MB (given our proofs are included). If our binary was light the relative difference in size would obviously be much more dramatic.

Reply:
That's a good observation. The difference in size does not seem so important. And perhaps if we cared about the size we can find many other optimizations and low-hanging fruits.

I'm personally ok with including debug symbols.